core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts

```diff
@@ -328,15 +328,23 @@ angular.module('oppia').directive('topicEditorTab', [
           };
 
           $scope.updatePracticeTabIsDisplayed = function(
-              newPracticeTabIsDisplayed) {
-            if (
-              newPracticeTabIsDisplayed !==
-              $scope.topic.getPracticeTabIsDisplayed()) {
-              TopicUpdateService.setPracticeTabIsDisplayed(
-                $scope.topic, newPracticeTabIsDisplayed);
+            newPracticeTabIsDisplayed) {
+          if (
+            newPracticeTabIsDisplayed !==
+            $scope.topic.getPracticeTabIsDisplayed() && $scope.hasMoreThanTenQuestions()) {
+            TopicUpdateService.setPracticeTabIsDisplayed(
+              $scope.topic, newPracticeTabIsDisplayed);
             }
           };
 
+          $scope.hasMoreThanTenQuestions = function() {
+            var countQuestions = 0;
+            $scope.topic.getSkillIds().forEach(item => {
+              countQuestions += $scope.skillQuestionCountDict[item];
+            })
+            return countQuestions >= 10;
```

Comment:
`$scope.skillQuestionCountDict` only includes the counts for the skills that are associated with the current topic. You can see that this is true because of the implementation in `topic-editor-state.service.ts`. So, you can simplify these lines to be:
```suggestion
            let skillQuestionCounts = Object.values($scope.skillQuestionCountDict);
            let numberOfPracticeQuestions = skillQuestionCounts.reduce((a, b) => a + b, 0);
            return numberOfPracticeQuestions >= 10;
```